python_modules/dagster/dagster/_config/pythonic_config/typing_utils.py

```diff
@@ -176,12 +173,21 @@ class FooResource(Resource):
     def __set_name__(self, _owner, name):
         self._assigned_name = name
 
-    def __get__(self: "T_Self", obj: Any, __owner: Any) -> "T_Self":
+    def __get__(self: Self, obj: Any, __owner: Any) -> Self:
         # no-op implementation (only used to affect type signature)
-        return cast(T_Self, getattr(obj, self._assigned_name))
-
-    def __set__(
-        self: "T_Self", obj: Optional[object], value: Union["T_Self", "PartialResource[T_Self]"]
-    ) -> None:
+        return cast(Self, getattr(obj, self._assigned_name))
+
+    # The annotation her has been temporarily changed from:
+    #     value: Union[Self, "PartialResource[Self]"]
+    # to:
+    #     value: Union[Any, "PartialResource[Any]"]
+    # This is because of a bug in mypy that is incorrectly interpreting the
+    # signature and can cause a false positive type error for users. This only
+    # started being detected in our test_type_signatures.py tests on 2024-02-02
+    # when some annotations elsewhere were added, likely causing mypy to
+    # analyze code it was previously skipping. The annotation should be
+    # reverted when the bug is fixed or another solution that surface as type
+    # errors for mypy users is found.
```

Comment:
how did you find this mypy issue?

Reply:
Some tests in `test_type_signatures.py` started failing. @benpankow and I investigated and it turned out that mypy and pyright had differing interpretations of the annotations.